pkg/agent/agent.go

```diff
@@ -761,7 +762,10 @@ func (i *Initializer) setupDefaultTunnelInterface() error {
 		externalIDs := map[string]interface{}{
 			interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaTunnel,
 		}
-		tunnelPortUUID, err := i.ovsBridgeClient.CreateTunnelPortExt(tunnelPortName, i.networkConfig.TunnelType, config.DefaultTunOFPort, shouldEnableCsum, localIPStr, "", "", "", nil, externalIDs)
+		extraOptions := map[string]interface{}{
+			"dst_port": strconv.Itoa(int(i.networkConfig.TunnelPort)),
+		}
```

Comment:
I think it should only set it when tunnelPort is not 0, otherwise it would really use 0 in my test:
```
32: genev_sys_0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 65000 qdisc noqueue master ovs-system state UNKNOWN group default qlen 1000
    link/ether 66:4c:3d:ae:e5:86 brd ff:ff:ff:ff:ff:ff
```